e2e/cat_test.go

```diff
@@ -0,0 +1,212 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"gotest.tools/v3/icmd"
+)
+
+func TestCatS3Object(t *testing.T) {
+	const (
+		bucket   = "bucket"
+		filename = "file.txt"
+	)
+
+	src := fmt.Sprintf("s3://%v/%v", bucket, filename)
+	contents, expected := getSequentialFileContent()
+
+	testcases := []struct {
+		name     string
+		cmd      []string
+		expected map[int]compareFunc
+	}{
+		{
+			name: "cat remote object",
+			cmd: []string{
+				"cat",
+				src,
+			},
+			expected: expected,
+		},
+		{
+			name: "cat remote object with json flag",
+			cmd: []string{
+				"--json",
+				"cat",
+				src,
+			},
+			expected: expected,
+		},
+		{
+			// generated file is ~61.670 MB, run with lower part size to assert sequential writes
+			name: "cat remote object with part-size set to 1MB",
+			cmd: []string{
+				"cat",
+				"-part-size",
+				"1",
+				src,
+			},
+			expected: expected,
+		},
+	}
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			s3client, s5cmd, cleanup := setup(t)
+			defer cleanup()
+
+			createBucket(t, s3client, bucket)
+			putFile(t, s3client, bucket, filename, contents)
+
+			cmd := s5cmd(tc.cmd...)
+			result := icmd.RunCmd(cmd)
+
+			result.Assert(t, icmd.Success)
+
+			assertLines(t, result.Stdout(), tc.expected)
+		})
+	}
+
+}
+
+func TestCatS3ObjectFail(t *testing.T) {
+	const (
+		bucket   = "bucket"
+		filename = "file.txt"
+	)
+
+	src := fmt.Sprintf("s3://%v/%v", bucket, filename)
+
+	testcases := []struct {
+		name     string
+		cmd      []string
+		expected map[int]compareFunc
+	}{
+		{
+			name: "cat non existent remote object",
+			cmd: []string{
+				"cat",
+				src,
+			},
+			expected: map[int]compareFunc{
+				0: contains(`ERROR "cat s3://bucket/file.txt": get returned with: NoSuchKey: status code: 404`),
+				1: equals(""),
+			},
+		},
+		{
+			name: "cat non existent remote object with json flag",
+			cmd: []string{
+				"--json",
+				"cat",
+				src,
+			},
+			expected: map[int]compareFunc{
+				0: contains(`{"operation":"cat","command":"cat s3://bucket/file.txt","error":"get returned with: NoSuchKey: status code: 404,`),
+				1: equals(""),
+			},
+		},
+		{
+			name: "cat remote object with glob",
+			cmd: []string{
+				"--json",
+				"cat",
+				src + "/*",
+			},
+			expected: map[int]compareFunc{
+				0: contains(`{"operation":"cat","command":"cat s3://bucket/file.txt/*","error":"remote source \"s3://bucket/file.txt/*\" can not contain glob characters"}`),
```

Comment:
We can use json() for testing json-formatted output. It supports multi-line strings.  Also `assertLines()` method can get  jsonCheck(bool) optional function for strict check. It validates if all lines at stout or stderr is json-decodable.

Reply:
Thanks, I missed that, but `json()` also replaces spaces in quotes.
Added `jsonCheck(true)` checks.